subsys/settings/settings_init.c

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright (c) 2018 Nordic Semiconductor ASA
+ * Copyright (c) 2015 Runtime Inc
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <assert.h>
+
+#include <string.h>
+#include <stdio.h>
+#include <stdbool.h>
+
+#include <errno.h>
+
+#include <settings/settings.h>
+#include <zephyr.h>
+
+void settings_init(void);
+extern int settings_backend_init(void);
+
+int settings_subsys_init(void)
+{
+	static bool settings_initialized;
+	int rc;
+
+	if (settings_initialized) {
```

Comment:
this check makes the "settings_cmd_inited" check in settings_init() not needed anymore